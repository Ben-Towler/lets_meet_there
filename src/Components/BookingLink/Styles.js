import styled from 'styled-components';

export const StyledBookingLink = styled.a`
  text-decoration: none;
  color: ${props => props.theme.colors.navyBlue};
  font-size: ${props => props.theme.fontSizes.small};
`;