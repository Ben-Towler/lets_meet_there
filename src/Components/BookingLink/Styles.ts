import styled from 'styled-components';

export const StyledBookingLink = styled.a`
`;