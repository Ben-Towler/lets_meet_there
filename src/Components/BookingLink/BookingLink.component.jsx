import React from 'react';
import { StyledBookingLink } from './Styles';
require('dotenv').config();

function BookingLink ({inbound, outbound, flight, city, favCity}) {

  function bookingParams () {
    const bookingDetails = [
      trim(city ? city : outbound.origin.CityName, 0, 4),
      trim(favCity ? favCity : inbound.origin.CityName, 0, 4),
      trim(flight.OutboundLeg.DepartureDate, 0, 10),
      trim(flight.InboundLeg.DepartureDate, 0, 10)
    ]

    return [...bookingDetails].join('/');
  }

  function trim (string, start, end) {
    return string.slice(start, end);
  }

  return (
    <StyledBookingLink
      href={ process.env.REACT_APP_BOOKING_URL + bookingParams() }
      rel="noopener noreferrer"
      aria-label="Book Flights"
      title="Book Flights"
      target="_blank"
    >
      Book Your Flights
    </StyledBookingLink>
  )
}

export default BookingLink;