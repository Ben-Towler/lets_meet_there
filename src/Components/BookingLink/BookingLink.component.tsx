import React from 'react';
import { StyledBookingLink } from './Styles';
import {Journey} from './BookingLink.test';
import {Flight as FlightInterface} from 'Interfaces/Flight'; 
require('dotenv').config();


interface Props {
  inbound: Journey;
  outbound: Journey;
  flight: FlightInterface;
  city?: string;
  favCity?: string;
}

function BookingLink ({inbound, outbound, flight, city, favCity}: Props): JSX.Element {
  
  function bookingParams () {
    const bookingDetails = [
      trim(city ? city : outbound.origin.CityName!, 0, 4),
      trim(favCity ? favCity : inbound.origin.CityName!, 0, 4),
      trim(flight.OutboundLeg.DepartureDate, 0, 10),
      trim(flight.InboundLeg.DepartureDate, 0, 10)
    ];
    return [...bookingDetails].join('/');
  }

  function trim (string: String, start: number, end: number) {
    return string.slice(start, end);
  }

  return (
    <StyledBookingLink
      href={ process.env.REACT_APP_BOOKING_URL + bookingParams() }
      rel="noopener noreferrer"
      aria-label="Book Flights"
      title="Book Flights"
      target="_blank"
    >
      Book Your Flights
    </StyledBookingLink>
  )
}

export default BookingLink;