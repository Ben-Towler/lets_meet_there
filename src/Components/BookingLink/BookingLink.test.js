import React from 'react';
import renderer from 'react-test-renderer';
import BookingLink from './BookingLink.component';
 
describe('Booking Link', () => {
  let inbound, outbound, flight;

  beforeEach(() => {
    inbound = {
      origin: {
        PlaceId: 51631,
        IataCode: 'FCO',
        Name: 'Rome Fiumicino',
        Type: 'Station',
        SkyscannerCode: 'FCO',
        CityName: 'Rome',
        CityId: 'ROME',
        CountryName: 'Italy'
      },
      destination: {
        PlaceId: 42414,
        IataCode: 'BCN',
        Name: 'Barcelona',
        Type: 'Station',
        SkyscannerCode: 'BCN',
        CityName: 'Barcelona',
        CityId: 'BARC',
        CountryName: 'Spain'
      }
    }

    outbound = {
      origin: {
        PlaceId: 65698,
        IataCode: 'LHR',
        Name: 'London Heathrow',
        Type: 'Station',
        SkyscannerCode: 'LHR',
        CityName: 'London',
        CityId: 'LOND',
        CountryName: 'United Kingdom'
      },
      destination: {
        PlaceId: 63238,
        IataCode: 'KEF',
        Name: 'Reykjavik Keflavik',
        Type: 'Station',
        SkyscannerCode: 'KEF',
        CityName: 'Reykjavik',
        CityId: 'REYK',
        CountryName: 'Iceland'
      }
    }

    flight = {
      QuoteId: 3,
      MinPrice: 1109,
      Direct: false,
      OutboundLeg: {
        CarrierIds: [
          1878
        ],
        OriginId: 42414,
        DestinationId: 48018,
        DepartureDate: '2020-06-24T00:00:00'
      },
      InboundLeg: {
        CarrierIds: [
          1324
        ],
        OriginId: 48018,
        DestinationId: 42414,
        DepartureDate: '2020-06-27T00:00:00'
      },
      QuoteDateTime: '2020-06-21T20:05:00'
    }
  });

  test('snapshot renders', () => {
    const component = renderer.create(
    <BookingLink 
      inbound={inbound}
      outbound={outbound}
      flight={flight}
    />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('snapshot renders', () => {
    const component = renderer.create(
    <BookingLink 
      inbound={inbound}
      outbound={outbound}
      flight={flight}
      city='Barcelona'
      favCity='Barcelona'
    />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});