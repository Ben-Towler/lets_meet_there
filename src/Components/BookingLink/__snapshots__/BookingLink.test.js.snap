// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking Link snapshot renders 1`] = `
<a
  aria-label="Book Flights"
  className="sc-AxjAm fDEmJb"
  href="https://www.skyscanner.net/transport/flights/Lond/Rome/2020-06-24/2020-06-27"
  rel="noopener noreferrer"
  target="_blank"
  title="Book Flights"
>
  Book Your Flights
</a>
`;

exports[`Booking Link snapshot renders 2`] = `
<a
  aria-label="Book Flights"
  className="sc-AxjAm fDEmJb"
  href="https://www.skyscanner.net/transport/flights/Barc/Barc/2020-06-24/2020-06-27"
  rel="noopener noreferrer"
  target="_blank"
  title="Book Flights"
>
  Book Your Flights
</a>
`;
